pkg/controller/argocd/keycloak.go

```diff
@@ -101,7 +117,7 @@ func getKeycloakContainer() corev1.Container {
 			{Name: "SSO_SERVICE_USERNAME", Value: "${SSO_SERVICE_USERNAME}"},
 			{Name: "SSO_SERVICE_PASSWORD", Value: "${SSO_SERVICE_PASSWORD}"},
 		},
-		Image:           "",
+		Image:           getKeycloakContainerImage(),
```

Comment:
Please inherit proxy env variables (`proxyEnvVars()`) from the operator (see `reconcileDexDeployment()` for example)

Reply:
Hi William, Adding proxyEnvVars() does not work with Keycloak. We need to add the Proxy settings instead to standalone-openshift.xml 

I have documented the steps on how to achieve this. 
https://github.com/argoproj-labs/argocd-operator/blob/c25532c31d4efc0a2258ffa005aecf14f3103981/docs/usage/keycloak.md#additional-steps-for-disconnected-openshift-clusters

Reply:
We inherit proxy env variables in all workloads.  The question is should we.   I understand that jboss add another layer to control policy to go out to proxy or not.   In case, there are other apps/processes running in the keycloak pod but on the jboss stack.  Having the proxy env var can make sure that the cluster proxy policy can be applied.

Reply:
RHSSO containers are dependent on "standalone-openshift.xml" for proxy mappings. 

Unfortunately the presence of "proxy env vars" somehow does not allow RHSSO container to read the entries in the "standalone-openshift.xml".  I have tested using both "proxy env vars" and "standalone-openshift.xml" but that does not work.

Reply:
If the presence of proxy env vars causes the request to OpenShift IdP not to go through proxy, it means that the proxy env vars DO have effect and the NO_PROXY env var causes the request (with the mapped/proxy address) not to go through the proxy.  That means the mapped/proxy address is found on NO_PROXY list.   It also tells me that if NO_PROXY were correctly configured (so that the request of the unmapped/IdP address would go through proxy), this mapping procedures (https://github.com/argoproj-labs/argocd-operator/blob/c25532c31d4efc0a2258ffa005aecf14f3103981/docs/usage/keycloak.md#additional-steps-for-disconnected-openshift-clusters) would not be needed.

Reply:
Requiring to manually perform the JBoss proxy mapping can be a problem for lot of customers.   Remember, you have to redo it when the pod is restarted.  :-(

Reply:
1. By setting both proxy env_vars and proxy-mappings in the standalone-openshift.xml, I can login to the Keycloak Pod and run the curl command to OAuth Server route and get the response back. I used the verbose option to confirm that the request is using "HTTPS_Proxy" and not "No_Proxy". Which means the HTTPS_Proxy env_var is well configured and NO_Proxy is not used.

2. Irrespective of misconfiguration of proxy env_vars, Configuring "Standalone-openshift.xml" is a must thing as per the Keycloak documentation and Keycloak folks.

I am equally confused on why adding proxy env_vars is causing an issue in this case but I am sure we cannot get rid of the manual steps to add the proxy_mappings in the "standalone-openshift.yaml"

I have the Image ready, we can manually add the proxy env_vars and test it.

Reply:
> Requiring to manually perform the JBoss proxy mapping can be a problem for lot of customers. Remember, you have to redo it when the pod is restarted. :-(

Totally agree on this , Keycloak team says this is the only way till they release v2.